lib/driver.js

```diff
@@ -411,11 +416,8 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.quit();
       await this.wda.uninstall();
       this.logEvent('wdaUninstalled');
-    } else if (!util.hasValue(this.wda.webDriverAgentUrl) && (await this.wda.isRunning())) {
-      log.info(`Will reuse previously cached WDA instance at '${this.wda.url.href}'. ` +
-               `Set the wdaLocalPort capability to a value different from ${this.wda.url.port} ` +
-               `if this is an undesired behavior.`);
-      this.wda.webDriverAgentUrl = this.wda.url.href;
+    } else if (!util.hasValue(this.wda.webDriverAgentUrl)) {
+      await this.wda.uninstallWDAIfRunningBundleIdIsDifferent(this.opts.updatedWDABundleId);
```

Comment:
`WDA` is not necessary in method name, since the class is already WebDriverAgent

Reply:
I'd rename the method to `setupCaching`